framework/src/base/FEProblem.C

```diff
@@ -2948,6 +3088,30 @@ FEProblem::setCouplingMatrix(CouplingMatrix * cm)
   _cm = cm;
 }
 
+void
+FEProblem::setNonlocalCouplingMatrix()
+{
+  unsigned int n_vars = _nl.nVariables();
+  _nonlocal_cm = new CouplingMatrix(n_vars);
```

Comment:
This does not work. `libmesh::CouplingMatrix` does not allow` CouplingMatrix _nonlocal_cm;` declaration. If we make it a reference then we have no way of initializing it in the constructor. I think pointer is the way to go.